math/dec.go

```diff
@@ -128,8 +128,9 @@ func (x Dec) Add(y Dec) (Dec, error) {
 func (x Dec) Sub(y Dec) (Dec, error) {
 	var z Dec
 	_, err := apd.BaseContext.Sub(&z.dec, &x.dec, &y.dec)
+
 	if err != nil {
-		return Dec{}, ErrInvalidDecString
+		return Dec{}, ErrInvalidDec.Wrap("exponent out of range: " + err.Error())
```

Comment:
Better not make assumptions about the error. Simple wrap provides enough context
```suggestion
		return Dec{}, ErrInvalidDec.Wrap(err.Error())
```